### PR TITLE
CEDS-1839 Refactor Additional Declaration Type into an Enumeration

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/forms/declaration/additionaldeclarationtype/AdditionalDeclarationType.scala [hunk 1]
```diff
@@ -16,12 +16,24 @@
...
+object AdditionalDeclarationType extends Enumeration {
+  type AdditionalDeclarationType = Value
+  implicit val format: Format[AdditionalDeclarationType.Value] =
+    Format(
+      Reads(
+        _.validate[String]
+          .filter(s => AdditionalDeclarationType.values.exists(_.toString == s))
+          .map(AdditionalDeclarationType.from(_).get)
+      ),
+      Writes(v => JsString(v.toString))
```
Comment (PatrykRudnicki) [inline app/forms/declaration/additionaldeclarationtype/AdditionalDeclarationType.scala]: Proposal: Change `v` to `value` and `s` to `string` or `type` or `input`

File: app/utils/validators/forms/FieldValidator.scala [hunk 2]
```diff
@@ -84,8 +84,8 @@ object FieldValidator {
 
   val startsWithCapitalLetter: String => Boolean = (input: String) => input.headOption.exists(_.isUpper)
 
-  val isContainedIn: Iterable[String] => String => Boolean =
-    (iterable: Iterable[String]) => (input: String) => iterable.exists(_ == input)
+  def isContainedIn[T](iterable: Iterable[T]): T => Boolean =
```
Comment (PatrykRudnicki) [inline app/utils/validators/forms/FieldValidator.scala]: Can we add some test that will cover different type? Actually we have tests just for string. It'll be good to have something for some different this that obvious type.

Comment (ed0906) [reply app/utils/validators/forms/FieldValidator.scala]: If it works for `String` surely it'd work for other types? it'd just be testing `iterable.exists(..)` works?

Comment (ed0906) [reply app/utils/validators/forms/FieldValidator.scala]: Actually there already is in the AdditionalDeclarationType form tests

Comment (ed0906) [reply app/utils/validators/forms/FieldValidator.scala]: As it first verifies what was entered was a valid enum value, then converts it to the enum, then verifies what was entered was a valid enum value for the options available.

Probably easier to look at the form mapping

**Pull request merged**